api/app/src/routes/medical/document.ts

```diff
@@ -30,16 +32,28 @@ const router = Router();
  * to be queried as well as the ones that have already been completed.
  *
  * @param req.query.patientId Patient ID for which to list documents.
+ * @param req.query.dateFrom Optional start date that docs will be filtered by (inclusive).
+ * @param req.query.dateTo Optional end date that docs will be filtered by (inclusive).
  * @return The available documents, including query status and progress - as applicable.
  */
 router.get(
   "/",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
     const patientId = getFromQueryOrFail("patientId", req);
+    const dateFrom = parseISODate(getFrom("query").optional("dateFrom", req));
+    const dateTo = parseISODate(getFrom("query").optional("dateTo", req));
 
     const documents = await getDocuments({ cxId, patientId });
-    const documentsDTO = toDTO(documents);
+    const documentsDTO = toDTO(documents).map(doc => {
+      if (!doc.indexed) return doc;
+      if (
+        (!dateFrom || dayjs(dateFrom).isBefore(doc.indexed)) &&
+        (!dateTo || dayjs(dateTo).isAfter(doc.indexed))
+      ) {
+        return doc;
+      }
+    });
```

Comment:
I assume we can't filter by indexed on DocumentReference?

Reply:
Yeah that's right, FHIR doesn't support this out of the box